src/Avalonia.Visuals/Animation/Animators/GradientBrushAnimator.cs

```diff
@@ -86,23 +91,31 @@ private IReadOnlyList<ImmutableGradientStop> InterpolateStops(double progress, I
             return stops;
         }
 
+        private ImmutableTransform InterpolateTransform(double progress, ITransform oldValue, ITransform newValue)
+        {
+            return new ImmutableTransform(_matrixAnimator.Interpolate(progress, oldValue.Value, newValue.Value));
```

Comment:
I wonder why for brush transform we can use way simpler matrix interpolation, while for control transform operations, we have way more complex implementation (that does not interpolate resulting matrix) - https://github.com/AvaloniaUI/Avalonia/blob/81e49c329e5b8461a780c41e7be525aaae0b570b/src/Avalonia.Visuals/Media/Transformation/TransformOperations.cs#L84

Is matrix interpolation less accurate? CC @MarchingCube

Reply:
I think that simply interpolating values of transform will not produce desired result (it might work for simplest transforms but for more complex ones it won't look good). Interpolation in control transform is implemented to follow css spec and retain order of operations if possible.

Reply:
![Animation](https://user-images.githubusercontent.com/2588062/128593781-b95d1b45-8dbb-45f2-b770-801c8559b0f6.gif)

Example of rotation from identity to a rotation of PI. Left is matrix interpolation from this PR, right one is transform operations interpolation.